crates/oxc_ast/src/ast/js.rs

```diff
@@ -1437,8 +1437,8 @@ pub struct BindingRestElement<'a> {
 /// Function Definitions
 #[visited_node]
 #[scope(
-    // TODO: `ScopeFlags::Function` is not correct if this is a `MethodDefinition`
-    flags(flags.unwrap_or(ScopeFlags::empty()) | ScopeFlags::Function),
+    // `flags` passed in to visitor via parameter defined by `#[visit_args(flags = ...)]` on parents
```

Comment:
To describe this I wanted to add something like this to the `visited_node`.

```rust
#[visited_node(args(flags: ScopeFlags))]
struct X;
```

It does absolutely nothing but express that this visited node has an extra argument called `flags` which is `ScopeFlags`. What do you think about it? I'm not sure if having this over a comment is beneficial so I'm looking forward to hearing from you.

Since with this marker, we still have to implement flags as an edge case(as we are doing right now) it might not be worth it and just add to the code noise.

I wasn't sure then and I'm not sure now.

Reply:
I would really like to find a way to remove this business with the extra arg entirely. It's the only place we do it.

If we had separate `FunctionDeclaration`, `FunctionExpression` and `ObjectMethod` types (along with class `MethodDefinition`), then each of them could hold the scope, and there'd be no need for this weird workaround. In my view, that'd also have other advantages.

So personally I think don't bother finessing this too much - let's just get rid of it entirely!

Reply:
https://github.com/oxc-project/backlog/issues/28 is relevant.